### DISCUSSION TITLE
(REQUEST) Loggin On/OFF

### DISCUSSION THREAD

**Discussion by mtklover** [FakeFishGames/Barotrauma#9601] [Ideas and suggestions]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

No bug report more of a request.

"This is whats destroying baro right now , the logs and or the horrible hosts/captains. I will bring up this conversation with the developer and see what can be achieved. Logs are useful yes but maybe make them available only on server lobby and or a On/Off Log button so it can be an optional thing on servers.
 
""What everyone is failing to talk about is when you find horrible hosts/captains that are absolute trash and deserve the griefing.
In a related note, i was talking with a mate how the logs ruin any chance at griefing/shenanigans. We have to accept trolling is a core part of the game as evidenced by the traitor feature. Logs make this impossible. Logs are not only completely unrealistic, they kill any relevance the security crew and cameras have. Logs destroy any attempt at deserved sabotage or mutiny when the host/captain is being a tyrant douche.
How do logs even know who did what if they managed to operate unseen? I believe logs should only be visible in the server lobby, and have a selectable option for them to be accessible in game. Maybe only enable them if you enable cheats in the server. Because in essence logs are a cheat.
I might make a discussion thread about logs.""

""Security is there to deal with griefers. Griefing is part of the game's lore and design. Mutiny is a real thing that was carried out when the captain deserved it. You say leaving the server is the "mature" thing to do? Mutiny is the real thing to do. Logs remove that option completely. Logs remove a whole aspect of gameplay that's both valid and and corresponds to the game setting and lore. You have not refuted any point I made. Renegades, husk worshippers, clowns, the insanity caused by living inside a tin can thousands of meters underwater surrounded by monsters that want to eat you. Every one of them is a lore reason to justify "griefing" and sabotage. Logs ruin any chance of this aspect of the game from manifesting. And even then i didn't suggest removing logs completely.""

### Reproduction steps

https://steamcommunity.com/app/602960/discussions/0/3052860266933074986/?ctp=4

### Bug prevalence

Happens every time I play

### Version

0.18.15.0

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Posting the same thing I posted in the Steam thread:

I mainly wanted to just comment on the criticism of the server logs, but I'll start with how I personally view griefing: I don't think it's the same thing as traitor/rogue gameplay, messing around, roleplaying someone who's gone crazy, etc. The latter can really spice up the rounds and be fun for everyone, while in my opinion griefing is by definition something that ruins the fun for others (or at least the majority of the server). And that I don't think is ok. It's also lazy, it takes a lot more creativity to play a clown, traitor, renegade or what have you in a way that creates interesting and fun gameplay for others as well.

...But to get to the logs. I can kind of understand the argument that the server log is a bad thing for the gameplay, but the way I see it, it's not a game mechanic, but a server moderation tool. It's not supposed to be realistic any more than being able to make you crew magically disappear by clicking a button that says "ban". In my view, the log _is_ a cheat, and you should not be reading it unless there's a very good reason to do so. For example, to find out out who's doing something that keeps ruining the fun for everyone (as opposed to falling within the "acceptable" limits of messing around).

This may be a bit of a blunt way to put it, but if you don't want to cheat by reading the logs, don't read them. Or if you don't want to play in a server where the host cheats by reading the logs, play in some other server. But I think taking useful (some might even say crucial) moderation tools away from server hosts and moderators would do more harm than good, even though they can be abused (as can basically any other tools the hosts and mods have access to).

**Comment by NilanthAnimosus** [comment]
I would just like to chime in that forcing this upon server hosts is essentially taking it away from player hosts as a vital tool, reducing their ability to moderate - and if they are using the server logs to ban players doing things they didn't want in a server they chose to host....even if terrible hosts, that is sort of their choice and right. using a toggle would also for serious hosts be like a self-gimping.

Punishing good hosts and server admins due to other bad hosts and server admins is not really an ideal solution to anything, logs removing an option to "Mutiny" can also be described as logs removing the easy option to wreck a server one doesn't like or doesn't intend to play normally on intended for a specific playstyle (IE. serious campaign or serious mission mode).

I could only accept this as a voluntary host toggle, but for most hosts this makes little sense over just not opening the server logs button, or being a mature enough host to not use that information to alter a round (Which if their not, they would not use a toggle anyways) unless they explicitly did not allow the actions to begin with (which on their hosted server, someone does something bad mid-round and they wish to ban said player...that honestly is their right as the server host given it is their server, the logs are just a tool available to hosts of any kind).

in my personal experience with hosting, I've always wanted more moderation tools, logging support and fair play. I would host cooperative and traitor modes, when I knew who was the valid traitors I took no action unless I saw it in-round directly as thats more fun for everyone in my server. what is not fun for everyone in my server if they want to play a round of traitor or cooperative mission/campaign, and some lunatic joins and in 5 minutes is unwiring all the ballasts or making the reactor explode. toggling off the logs visibility would not make a reasonable host happier or better at hosting somehow, and just (in my opinion with all the bad actors, players and hosts alike) degrade everybody's experience except the trolls and griefers...which making barotrauma closer to an unregulated wild west certainly wouldn't make a player or host like myself happy at all.

If a host wishes to have a more wild server where griefers are allowed or banned post-round, they can just check the logs post round or check mid-round who is the bad actor, but ban post-round.

so I think mid round viewing of logs should not be taken away, and a toggle for such would not be used by those you don't want to view it (the "Bad hosts" you want to mutiny against or not "cheat" in rounds) or mature/serious hosts that can prevent themselves acting on or looking at information from said logs for their own or their players fun. so however implemented it would be only a detriment or pointless unused feature in my eyes.

I would finally like to point out - players can host their servers as a type intended for total chaos, or indicate griefing is allowed. host freedom already exists to host servers of the type they want and communicate that in the server list somewhat, the logs merely a tool anyone can use for the good or bad.
